tests/test_integration.py

```diff
@@ -86,6 +92,44 @@ def test_train_and_run(
     assert psms.loc[4, "sequence"] == "PEPTLDEK"
     assert psms.loc[4, "spectra_ref"] == "ms_run[2]:scan=111"
 
+    # Test checkpoint saving when val_check_interval is greater than training steps
```

Comment:
**suggestion:** I would split this off in a dedicated test to indicate that you're really only testing the model saving part here.

I also think that this is more of a unit test than an integration test, so I would recommend to execute this as `ModelRunner.train` insted of through the CLI runner..